conventions/README.md

```diff
@@ -271,3 +271,49 @@ focusMenu(): void => {
   this.menuFocusTimeout = window.setTimeout(() => focusElement(this.menuEl), 100);
 }
 ```
+
+## Utils
+
+There are utilities for common workflows in [`src/utils`](https://github.com/Esri/calcite-components/tree/master/src/utils).
+
+### Global attributes
+
+The [`globalAttributes`](https://github.com/Esri/calcite-components/blob/master/src/utils/globalAttributes.ts) util was specifically made to access the `lang` global attribute when set on a Calcite component. However, it can be extended to allow additional global attributes by adding to the [`allowedGlobalAttributes`](https://github.com/Esri/calcite-components/blob/a33aa0df0c5bf103f91187826e6b12b8ff266d90/src/utils/globalAttributes.ts#L4-L5) array. The util is used in [`calcite-pagination`](https://github.com/Esri/calcite-components/blob/master/src/components/pagination/pagination.tsx), which you can use as a reference.
+
+#### Usage steps
+
+1. Import the interface and watch/unwatch methods
+
+   ```js
+   import { GlobalAttrComponent, watchGlobalAttributes, unwatchGlobalAttributes } from "../../utils/globalAttributes";
+   ```
+
+2. Implement the interface
+
+   ```js
+   export class ComponentName implements GlobalAttrComponent {
+   ```
+
+3. Add `globalAttributes` state
+
+   ```js
+   @State() globalAttributes = {};
+   ```
+
+4. Add connect/disconnect callbacks
+
+   ```js
+   connectedCallback(): void {
+       watchGlobalAttributes(this, ["lang"]);
+   }
+
+   disconnectedCallback(): void {
+     unwatchGlobalAttributes(this);
+   }
+   ```
+
+5. Use the state to access `lang` (or another global attribute that may be allowed in the future).
+
+   ```js
+   const lang = this.globalAttributes["lang"] || document.documentElement.lang || "en";
```

Comment:
I wonder if we should have a utility for this part: `document.documentElement.lang || "en"`. I think we want consistency here and it may actually be better as the following:

```ts
const lang = this.globalAttributes["lang"] || this.el.closest("lang") || "en"; // may want to do `closestElementCrossShadowBoundary` instead?
```

Reply:
I completely agree, you actually said something similar in my [PR](https://github.com/Esri/calcite-components/pull/4574#discussion_r876069296) for implementing the util in pagination. Should I create a refactor issue for it? In the issue I can mention to change that line of the convention depending on the solution. Or if its a quick refactor I can hold off on merging until it happens.

Reply:
Oh nvm, you already made a refactor issue #4576